src/codegate/providers/ollama/provider.py

```diff
@@ -46,12 +46,42 @@ def _setup_routes(self):
         Sets up Ollama API routes.
         """
 
+        @self.router.get(f"/{self.provider_route_name}/api/tags")
+        async def get_tags(request: Request):
+            """
+            Special route for /api/tags that responds outside of the pipeline
+            Tags are used to get the list of models
+            https://github.com/ollama/ollama/blob/main/docs/api.md#list-local-models
+            """
+            async with httpx.AsyncClient() as client:
+                response = await client.get(f"{self.base_url}/api/tags")
+                return response.json()
+
+        @self.router.post(f"/{self.provider_route_name}/api/show")
```

Comment:
this seems unrelated to cline support?

Reply:
The route or use of ollama?

Reply:
ah i get it now, you add it because cline is needing additional endpoints that were not exposed before. I hit the same with Aider